WordPress/gutenberg
# Block Editor: Simplify `BlockHTMLConvertButton`

## What?
This PR makes a few improvements to the `BlockHTMLConvertButton` component:

## Why?
A micro-optimization that makes the rendered component tree smaller, but also makes that part of the code more readable.

## How?
This PR:
* Removes the nested `BlockConvertButton` which is used only once anyway.
* Simplifies logic that is unnecessarily split into two components and condenses it into one.
* Replaces `withSelect` with `useSelect`, `withDispatch` with `useDispatch` and removes the unnecessary `compose`, making the component tree smaller and more readable.
* Removes the unnecessary `small` prop that wasn't used.

## Testing Instructions
* Insert an HTML block with some HTML in it. 
* Click the block settings menu of the HTML block.
* Verify the "Convert to Blocks" item is there and it still works well.
* Verify that that item does not appear if this is not an HTML block.

### Testing Instructions for Keyboard
None.

## Screenshots or screencast 
Component tree before:
![Screenshot 2023-10-02 at 16 48 40](https://github.com/WordPress/gutenberg/assets/8436925/0d9a79ce-2a69-46d3-b7a3-92222b57812d)


Component tree after:
![Screenshot 2023-10-02 at 16 49 29](https://github.com/WordPress/gutenberg/assets/8436925/20dbf151-4f85-45b7-a8cd-e536f3d726bf)
